src/demo/layout/MainLayout.tsx

```diff
@@ -11,58 +10,42 @@ import SwapHorizOutlinedIcon from "@mui/icons-material/SwapHorizOutlined";
 import ChatBubbleOutlineOutlinedIcon from "@mui/icons-material/ChatBubbleOutlineOutlined";
 import LogoutIcon from "@mui/icons-material/Logout";
 import logo from "../assets/tonomy-logo48.png";
-import "./main.css";
+import { useUserStore } from "../../common/stores/user.store";
+import "./MainLayout.css";
 
-const MainLayout = ({ content }) => {
+export type MainLayoutProps = {
+  onLogout: () => void;
+};
+
+const MainLayout = (props: MainLayoutProps) => {
   const { collapseSidebar } = useProSidebar();
+  const [collapse, setCollapse] = useState(false);
   const [user, setUser] = useState<ExternalUser | null>(null);
   const navigation = useNavigate();
+  const userStore = useUserStore();
 
   async function onRender() {
-    try {
-      const user = await api.ExternalUser.getUser();
+    const user = await api.ExternalUser.getUser();
```

Comment:
instead,
```ts
const user = userStore.user
if (!user) navigate back to login page
```
this will get it from the user store, which is initialized in the login or the callback page not here